src/Templating/TemplateLocator.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Templating;
+
+use Symfony\Bundle\FrameworkBundle\Templating\Loader\TemplateLocator as FrameworkTemplateLocator;
+
+/**
+ * @deprecated since 3.9, to be removed with 4.0.
```

Comment:
why? its new

Reply:
I've created new service `sonata.templating`, to start migrating from `templating` to `twig` service without BC breaks in 3.x branch. `sonata.templating`, like `templating` service, has 3 arguments: `twig`, `sonata.templating.name_parser` and `sonata.templating.locator`. This new class is `sonata.templating.locator` service.

It is a new class, yes. It is needed in 3.x branch to be sure, that everything will be fine after deprecating of `symfony/templating` component. But this new class should not be used by other services, because it will be deleted in master branch (if PR will be merged, of course) accordingly with stage 2 of a plan, described in #456 

So, it is new and deprecated at the same time.